src/chainstate/stacks/db/blocks.rs

```diff
@@ -6047,6 +6047,19 @@ impl StacksChainState {
         query_row(&self.db(), sql, args).map_err(Error::DBError)
     }
 
+    pub fn get_stacks_chain_parent(
+        &self,
+        sortdb: &SortitionDB,
+        staging_block: &StagingBlock,
+    ) -> Result<Option<StagingBlock>, Error> {
```

Comment:
Can you call this `get_stacks_block_parent()` instead of `get_stacks_chain_parent()`?  You're returning a block, not a chain ;)